### DISCUSSION TITLE
About Atari game display problem

### DISCUSSION THREAD

**Discussion by yupingyuping** [community/community#34402] [Other Feature Feedback, Questions, & Ideas]
I want to display an atari game by env.render("rgb-array"). 
The evn:" ID = 'BreakoutNoFrameskip-v4'  env = gym.make(ID)" 
But it doesn't work
my installation version are: python 3.9.12, tensorflow 2.9.1,ale_py0.7.5 keras 2.9.0
In fact I can save suc the pictures in every episode successfully.So I don't think I missed any installation. But I cann't show the graphics like Cartpole,Mountain Car。
Need anyone's help
Many Thanks

Attachment are the picture, later are my codes. 
![2](https://user-images.githubusercontent.com/114548798/192677764-973fea27-63f5-479a-b938-f17fd258f8d3.png)

import atari_py
from skimage import io
print(atari_py.list_games())

import time
import gym
import matplotlib.pyplot as plt

ID = 'BreakoutNoFrameskip-v4'
env = gym.make(ID)
# env = gym.make('ALE/Breakout-v4')
print("Observation Space: ", env.observation_space)
print("Action Space       ", env.action_space)
obs = env.reset()

for i in range(10):
    env.render("rgb_array") #(ID, render_mode = "human")
    action = env.action_space.sample()
    obs, reward, done, info = env.step(action)
    jpgname = './pici23-%d.jpg' % i
    io.imsave(jpgname, obs)
    time.sleep(0.01)
env.close()

**Comment by yupingyuping** [comment]
I had print the evn by print(gym.envs.registry.all()), There are BreakoutNoFrameskip-v4   ALE\Breakout-v4 and other atari geme registered

**Comment by dennybill** [comment]
It seems that you are trying to display an Atari game using env.render("rgb-array"), but it is not working. You have mentioned that you can save the pictures in every episode successfully, but you are unable to show the graphics like Cartpole or Mountain Car.

One possible reason for this could be that the Atari game is not compatible with the render() function. The render() function can display images for some environments, but not all. In some cases, you may need to use a different method to display the game.

One approach you could try is to use the gym.wrappers.Monitor class to record the game as a video and then play it back using a video [rent](https://swishlimohouston.com/) car player. Here is an example code snippet that demonstrates how to use the Monitor class:

python
Copy code
import gym
from gym.wrappers import Monitor

ID = 'BreakoutNoFrameskip-v4'
env = gym.make(ID)

# wrap the environment with Monitor to record video
env = Monitor(env, './videos', force=True)

obs = env.reset()

for i in range(10):
    env.render()
    action = env.action_space.sample()
    obs, reward, done, info = env.step(action)

env.close()
This code will record a video of the game and save it in the ./videos directory. You can then play the video using a video player.

Another approach you could try is to use the pyglet library to display the game. Here is an example code snippet that demonstrates how to use pyglet:

python
Copy code
import gym
import pyglet

ID = 'BreakoutNoFrameskip-v4'
env = gym.make(ID)

window = pyglet.window.Window()

obs = env.reset()

@window.event
def on_draw():
    window.clear()
    pyglet.image.ImageData(
        env.observation_space.shape[0],
        env.observation_space.shape[1],
        'RGB',
        obs.tobytes(),
    ).blit(0, 0)

def update(dt):
    action = env.action_space.sample()
    obs, reward, done, info = env.step(action)
    if done:
        pyglet.app.exit()

pyglet.clock.schedule_interval(update, 1/60.0)
pyglet.app.run()
This code will display the game using a Pyglet window. You can use the arrow keys to control the paddle in the game. Note that you may need to install the pyglet library using pip install pyglet.

I hope these suggestions help you display the Atari game successfully.
